shared-libs/transitions/test/unit/transitions.js

```diff
@@ -333,5 +336,186 @@ describe('transitions', () => {
+
+      const result = await new Promise((resolve, reject) => {
+        transitions.applyTransition(
+          { key: transition.key, change, transition },
+          (err, result) => err ? reject(err) : resolve(result)
+        );
+      });
+
+      expect(result).to.equal(undefined);
+      expect(transition.filter.args).to.deep.equal([[change.doc, change.info]]);
+      expect(transition.onMatch.called).to.equal(false);
+      expect(infodoc.updateTransition.called).to.equal(false);
+    });
+
+    it('should skip transitions that cant run', async () => {
+      const transition = {
+        key: 'mytransition',
+        filter: sinon.stub(),
+        onMatch: sinon.stub(),
+      };
+
+      sinon.stub(infodoc, 'updateTransition');
+      const change = { deleted: true, doc: {}, info: {} };
+
+      const result = await new Promise((resolve, reject) => {
+        transitions.applyTransition(
+          { key: transition.key, change, transition },
+          (err, result) => err ? reject(err) : resolve(result)
+        );
+      });
+
+      expect(result).to.equal(undefined);
+      expect(transition.filter.called).to.equal(false);
+      expect(transition.onMatch.called).to.equal(false);
+      expect(infodoc.updateTransition.called).to.equal(false);
+    });
+
+    it('should force running a transition', async () => {
+      const transition = {
+        key: 'mytransition',
+        filter: sinon.stub(),
+        onMatch: sinon.stub().resolves(true),
+      };
+
+      sinon.stub(transitions, 'canRun').returns(false);
+      sinon.stub(infodoc, 'updateTransition');
+      const change = { doc: {}, info: {} };
+
+      const result = await new Promise((resolve, reject) => {
+        transitions.applyTransition(
+          { key: transition.key, change, transition, force: true },
+          (err, result) => err ? reject(err) : resolve(result)
+        );
+      });
+
+      expect(result).to.equal(true);
+      expect(transitions.canRun.called).to.equal(false);
+      expect(transition.onMatch.args).to.deep.equal([[change]]);
+      expect(infodoc.updateTransition.args).to.deep.equal([[change, transition.key, true]]);
+    });
+
+    it('should catch transition filter errors', async () => {
+      const transition = {
+        key: 'mytransition',
+        filter: sinon.stub().throws(new Error('boom')),
+        onMatch: sinon.stub(),
+      };
+      sinon.stub(infodoc, 'updateTransition');
+      const change = { doc: {}, info: {} };
+
+      const result = await new Promise((resolve, reject) => {
+        transitions.applyTransition(
+          { key: transition.key, change, transition },
+          (err, result) => err ? reject(err) : resolve(result)
+        );
+      });
+      expect(result).to.equal(false);
+      expect(transition.filter.args).to.deep.equal([[change.doc, change.info]]);
+      expect(transition.onMatch.called).to.equal(false);
+      expect(infodoc.updateTransition.called).to.equal(false);
+    });
```

Comment:
Do we typically try to include assertions for the error logging?

Reply:
I think it depends on the case, and so far it comes down to preference, and whether the logged errors are really imporant: do they make it to a feedback doc?, does the client see them?, etc. 
In this case, for a patch release that I might need to backport 3 versions back, I wanted to reduce the risk of a more complex cherry-pick.